src/main/java/com/volmit/iris/util/decree/DecreeNode.java

```diff
@@ -47,13 +47,21 @@ public DecreeNode(Object instance, Method method) {
      * @return The list of parameters if ALL are annotated by @{@link Param}, else null
      */
     public KList<DecreeParameter> getParameters() {
-        KList<DecreeParameter> p = new KList<>();
+        KList<DecreeParameter> required = new KList<>();
+        KList<DecreeParameter> optional = new KList<>();
```

Comment:
No, undo this entire file

Reply:
Why not put required params before non-required?